precompiles/ics02/ics02.go

```diff
@@ -0,0 +1,148 @@
+package ics02
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	_ "embed"
+
+	ibcutils "github.com/cosmos/evm/ibc"
+	cmn "github.com/cosmos/evm/precompiles/common"
+	evmtypes "github.com/cosmos/evm/x/vm/types"
+
+	"cosmossdk.io/log"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ vm.PrecompiledContract = (*Precompile)(nil)
+
+const (
+	// TODO: These gas values are placeholders and should be determined through proper benchmarking.
```

Comment:
I put my reduced estimation. I can actually benchmark solidity IBC Eureka, in which case the tx would be around ~300,000 due to zk proofs there